pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -74,5 +81,122 @@ func (r *ReconcileChangeTierRequest) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	// if is complete, then we can delete it
+	if condition.IsTrue(changeTierRequest.Status.Conditions, toolchainv1alpha1.ChangeTierRequestComplete) {
+		return reconcile.Result{}, r.changeIsComplete(changeTierRequest)
+	}
+
+	err = r.changeTier(reqLogger, changeTierRequest, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, r.changeIsComplete(changeTierRequest)
+}
+
+func (r *ReconcileChangeTierRequest) changeIsComplete(changeTierRequest *toolchainv1alpha1.ChangeTierRequest) error {
```

Comment:
Or maybe just complete (so it sounds more like an action)?
```suggestion
func (r *ReconcileChangeTierRequest) complete(changeTierRequest *toolchainv1alpha1.ChangeTierRequest) error {
```

Reply:
sure, why not :-) 88e9fad